examples/fluids/problems/stg_shur14.c

```diff
@@ -486,3 +488,24 @@ PetscErrorCode SetupStrongSTG(DM dm, SimpleBC bc, ProblemData *problem) {
 
   PetscFunctionReturn(0);
 }
+
+PetscErrorCode SetupStrongSTG_QF(Ceed ceed, ProblemData *problem,
+                                 CeedInt num_comp_x, CeedInt num_comp_q, CeedInt q_data_size_sur,
+                                 CeedQFunction *pqf_strongbc) {
+
+  CeedQFunction qf_strongbc;
+  CeedQFunctionContext stg_context=NULL;
+  PetscFunctionBeginUser;
+  CeedQFunctionCreateInterior(ceed, 1, STGShur14_Inflow_StrongQF,
+                              STGShur14_Inflow_StrongQF_loc, &qf_strongbc);
+  CeedQFunctionAddInput(qf_strongbc, "surface qdata", q_data_size_sur,
+                        CEED_EVAL_NONE);
+  CeedQFunctionAddInput(qf_strongbc,  "x",     num_comp_x, CEED_EVAL_NONE);
+  CeedQFunctionAddInput(qf_strongbc,  "scale", 1,          CEED_EVAL_NONE);
+  CeedQFunctionAddOutput(qf_strongbc, "q",     num_comp_q, CEED_EVAL_NONE);
+
+  CeedQFunctionContextReferenceCopy(problem->ics.qfunction_context, &stg_context);
```

Comment:
I've run the code through valgrind, but am not sure how to interpret the output with regard to seeing if a specific context is destroyed. Running with `--leak-check=full --show-leak-kinds=all --track-origins=yes`, and the only references to contexts are for the creation of the STG context, but not this specific line.

Reply:
If it's still showing as allocated at the end, that means the destructor wasn't called enough times. I suspect it needs another destory here. Alternatively you could directly set the context for the QF from the problem struct instead of copying the reference an additional time.

Reply:
> If it's still showing as allocated at the end, that means the destructor wasn't called enough times.

I'm not sure how to determine that based on the valgrind output. 

> I suspect it needs another destory here.

Yeah, I think that's right. I didn't realize that the context set to the `problem->apply_<bc>.qfunction_context`s was destroyed after being set, so I misunderstood the lifecycle of the QFContexts.